codersdk/deployment.go

```diff
@@ -33,6 +33,21 @@ const (
 	EntitlementNotEntitled Entitlement = "not_entitled"
 )
 
+// entitlementWeight converts the enum types to a numerical value for easier
+// comparisons. Easier than sets of if statements.
+func entitlementWeight(e Entitlement) int {
```

Comment:
Might be cleaner to add this as a func on `Entitlement` type itself. `e.Weight()`